priv/nodejs/pyramid-tiff/pyramid.js

```diff
@@ -1,81 +1,65 @@
 const AWS = require("aws-sdk");
+const concat = require("concat-stream");
 const sharp = require("sharp");
 const URI = require("uri-js");
-const fs = require("fs");
-const tempy = require("tempy");
+const s3 = new AWS.S3();
 
-const createPyramidTiff = async (source, dest) => {
-  const inputFile = await makeInputFile(source);
-  try {
-    console.info(`Creating pyramidal TIFF from ${inputFile}`);
-    const pyramidTiff = await sharp(inputFile, { limitInputPixels: false })
-      .resize({
-        width: 15000,
-        height: 15000,
-        fit: "inside",
-        withoutEnlargement: true
-      })
-      .tiff({
-        compression: "jpeg",
-        quality: 75,
-        tile: true,
-        tileHeight: 256,
-        tileWidth: 256,
-        pyramid: true
-      })
-      .toBuffer();
-    await sendToDestination(pyramidTiff, dest);
-    return dest;
-  } finally {
-    console.info(`Deleting ${inputFile}`);
-    fs.unlink(inputFile, err => {
-      if (err) {
-        throw err;
-      }
-    });
-  }
-};
-
-const makeInputFile = location => {
+const createPyramidTiff = (source, dest) => {
   return new Promise((resolve, reject) => {
-    let uri = URI.parse(location);
-    let fileName = tempy.file();
-    console.info(`Retrieving ${location} to ${fileName}`);
-    let writable = fs
-      .createWriteStream(fileName)
-      .on("error", err => reject(err));
-    let s3Stream = new AWS.S3()
-      .getObject({
-        Bucket: uri.host,
-        Key: getS3Key(uri)
-      }).createReadStream();
+    try {
+      console.log(`Creating pyramid from ${source}`);
+      const inputStream = streamFromS3(source);
+
+      let metadata;
+      const transformStream = sharp({ limitInputPixels: false })
```

Comment:
Should we test chaining the `removeAlpha` operation here as well? We've been bitten by alpha channels many times, and I just came across it in the [docs](https://sharp.pixelplumbing.com/api-channel#removealpha).